### PR TITLE
Update woocommerce_suggest_jetpack filter description

### PR DESCRIPTION
**Pull request by chihsuan**
###_ Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR updates `woocommerce_suggest_jetpack` filter description as suggested in  https://github.com/woocommerce/woocommerce/pull/38286#discussion_r1202992432

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Review the changes

### CODE REVIEW

File: plugins/woocommerce/src/Admin/API/OnboardingFreeExtensions.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function get_items_permissions_check( $request ) {
 	public function get_available_extensions( $request ) {
 		$extensions = RemoteFreeExtensions::get_extensions();
 		/**
-		* Remove Jetpack when woocommerce_suggest_jetpack is false.
+		* When false, Jetpack is removed from the list of suggested extensions.
```
Comment (rjchow) [inline plugins/woocommerce/src/Admin/API/OnboardingFreeExtensions.php]:
```suggestion
		* When false, Jetpack is removed from the list of extensions suggested in the Onboarding Profiler. This list is first retrieved from the WooCommerce.com API, then if a plugin with the 'jetpack' slug is found, it is removed.
```

**Review by rjchow — COMMENTED** [review]
Thanks! I've added a few words for context

File: plugins/woocommerce/src/Admin/API/OnboardingFreeExtensions.php (hunk 1 above)
Comment (adrianduffell) [reply plugins/woocommerce/src/Admin/API/OnboardingFreeExtensions.php]: Love the improvements here! One thing to note - I think we should eventually apply the filter across all of WC Admin where Jetpack is suggested. It is also in Analytics for example ([screenshot in this issue](https://github.com/woocommerce/woocommerce/issues/37016)).

To iterate a bit on everyone's ideas, my suggestion is:

```
* Allows removing Jetpack suggestions from WooCommerce Admin when false.
*
* In this instance it is removed from the list of extensions suggested in the Onboarding Profiler. This list is first retrieved from the WooCommerce.com API, then if a plugin with the 'jetpack' slug is found, it is removed
```

Comment (chihsuan) [reply plugins/woocommerce/src/Admin/API/OnboardingFreeExtensions.php]: Thank you! 😄  @adrianduffell @rjchow Updated in e84f289b1dc11fadf5d304a5bf5069e49246af16.

**Review by rjchow — APPROVED** [review]
😀

**Review by adrianduffell — APPROVED** [review]
LGTM! 🚀 

Thank you @chihsuan @rjchow !

**Pull request merged**